service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java

```diff
@@ -0,0 +1,61 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import java.io.IOException;
+import java.time.Instant;
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.api.ResourcesApi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+
+public class SamResourceClient {
+  private final String samResourceId;
+  private final TokenChecker tokenChecker;
+  private final ApiClient samClient;
+
+  private final String SAM_RESOURCE_TYPE = "controlled-application-private-workspace-resource";
+
+  private final Logger logger = LoggerFactory.getLogger(SamResourceClient.class);
+
+  public SamResourceClient(String samResourceId, ApiClient samClient, TokenChecker tokenChecker) {
+    this.samResourceId = samResourceId;
+    this.tokenChecker = tokenChecker;
+    this.samClient = samClient;
+  }
+
+  // Should only be used in checkCachedPermission, but making it public so that we can test it
+  @Cacheable("expiresAt")
+  public Instant checkWritePermission(String accessToken) {
+    try {
+      var oauthInfo = tokenChecker.getOauthInfo(accessToken);
+      if(oauthInfo.expiresAt().isPresent()) {
```

Comment:
Google code style also requires a space between `if` and `(`. This is to help visually distinguish function calls from non-function calls.
```suggestion
      if (oauthInfo.expiresAt().isPresent()) {
```

Reply:
I'll leave all the formatting to spotless, which is currently broken due to java 17 upgrade...think we can leave that out for this PR

Reply:
appreciate it if you have some idea how to fix spotless tho

Reply:
I have a fix for the spotless config in a branch off of this branch, but can't commit to this repo. If you give me write privs, I'll push the branch to PR into this one.